app/facades/facet_configuration_facade.rb

```diff
@@ -27,10 +27,20 @@ def all_solr_fields
 
   def solr_field_iterator
     @solr_field_iterator ||= iterator_factory.new(all_solr_fields).map do |k, v|
+      setup_facet_item_set(v) if k == :set
       [k, HasFacetField.new(v, grouped_facets[k].try(:first) || FacetField.new(:key => k))]
     end
   end
 
+  def setup_facet_item_set(value)
+    top_terms = value.topTerms.each_slice(2).to_a
+    top_terms.each { |item| facet_item item }
+  end
+
+  def facet_item(item)
+    FacetItem.where(:value => item.first).first_or_create
```

Comment:
It finds an instance of FacetItem in the table, and creates one if it doesn't exist yet.